cloud/linode/instances_test.go

```diff
@@ -33,50 +31,36 @@ func TestInstanceExists(t *testing.T) {
 	defer ctrl.Finish()
 
 	client := NewMockClient(ctrl)
-	instances := newInstances(client)
-
-	t.Run("should propagate generic api error", func(t *testing.T) {
-		node := nodeWithProviderID(providerIDPrefix + "123")
-		expectedErr := errors.New("some error")
-		client.EXPECT().GetInstance(gomock.Any(), 123).Times(1).Return(nil, expectedErr)
-
-		exists, err := instances.InstanceExists(ctx, node)
-		assert.ErrorIs(t, err, expectedErr)
-		assert.False(t, exists)
-	})
 
 	t.Run("should return false if linode does not exist (by providerID)", func(t *testing.T) {
+		instances := newInstances(client)
 		node := nodeWithProviderID(providerIDPrefix + "123")
-		client.EXPECT().GetInstance(gomock.Any(), 123).Times(1).Return(nil, &linodego.Error{
-			Code: http.StatusNotFound,
-		})
+		client.EXPECT().ListInstances(gomock.Any(), nil).Times(1).Return([]linodego.Instance{}, nil)
 
 		exists, err := instances.InstanceExists(ctx, node)
 		assert.NoError(t, err)
 		assert.False(t, exists)
 	})
 
 	t.Run("should return false if linode does not exist (by name)", func(t *testing.T) {
+		instances := newInstances(client)
 		name := "some-name"
 		node := nodeWithName(name)
-		notFound := &linodego.Error{
```

Comment:
The nice thing about this implementation is that we no longer use http-specific errors where in practice we don't care about HTTP issues, we just want to retrieve some data.